actors/builtin/miner/miner_state.go

```diff
@@ -821,8 +822,12 @@ func (st *State) UnlockUnvestedFunds(store adt.Store, currEpoch abi.ChainEpoch,
 	err = vestingFunds.ForEach(&lockedEntry, func(k int64) error {
 		if amountUnlocked.LessThan(target) {
 			if k >= int64(currEpoch) {
-				unlockAmount := big.Max(big.Sub(target, amountUnlocked), lockedEntry)
+				unlockAmount := big.Min(big.Sub(target, amountUnlocked), lockedEntry)
+
 				lockedEntry = big.Sub(lockedEntry, unlockAmount)
+				if err := vestingFunds.Set(uint64(k), &lockedEntry); err != nil {
+					return err
+				}
```

Comment:
Let's put this inside an `else` clause after line 833, so we either update or delete it.